azext_iot/operations/central.py

```diff
@@ -18,17 +17,28 @@ def find_between(s, start, end):
 
 
 def iot_central_device_show(
-    cmd, device_id, app_id, central_api_uri="api.azureiotcentral.com"
+    cmd, device_id, app_id, central_api_uri="azureiotcentral.com"
 ):
-    sasToken = get_iot_hub_token_from_central_app_id(cmd, app_id, central_api_uri)
-    endpoint = find_between(sasToken, "SharedAccessSignature sr=", "&sig=")
-    target = {"entity": endpoint}
-    auth = BasicSasTokenAuthentication(sas_token=sasToken)
-    service_sdk, errors = _bind_sdk(target, SdkType.service_sdk, auth=auth)
-    try:
-        return service_sdk.get_twin(device_id)
-    except errors.CloudError as e:
-        raise CLIError(unpack_msrest_error(e))
+    from azext_iot.common._azure import get_iot_central_tokens
+
+    tokens = get_iot_central_tokens(cmd, app_id, central_api_uri)
+    exception = None
+
+    # The device could be in any hub associated with the given app.
+    # We must search through each IoT Hub until device is found.
```

Comment:
is there a pointed-lookup way of determining the iothub for a device? 

I'm currently in the middle of a PR that could help you get the IoTHub given a device_id, id_scope and attestation key using the global dps API.

Not sure how high pri this PR is, but if you are willing to wait I think reusing my code might be good

Reply:
Spoke to @prbans  offline.  This is not possible right now. Resolving